### PR TITLE
Migrate S3834 to eslint implementation (no-new-symbol)

### LINKED ISSUE #2109: Migrate S3834 to eslint frontend for JS

### PR DESCRIPTION
**Pull request by yassin-kammoun-sonarsource**
Fixes #2109

### CODE REVIEW

File: eslint-bridge/tests/rules/no-new-symbol.test.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  ],
+  invalid: [
+    {
+      code: `
+        var sym =
+          new Symbol("foo");`,
+      errors: [
+        {
+          message:
+            '{"message":"Remove this \\"new\\" operator.","secondaryLocations":[{"column":10,"line":3,"endColumn":27,"endLine":3}]}',
```
Comment (andrea-guarino-sonarsource) [inline eslint-bridge/tests/rules/no-new-symbol.test.ts:84]: Aren't primary and secondary location overlapping? Do we really need a secondary location?

**Review by andrea-guarino-sonarsource — COMMENTED** [review]
Isn't it valid also for typescript?

File: eslint-bridge/tests/rules/no-new-symbol.test.ts (hunk 1 above)
Comment (yassin-kammoun-sonarsource) [reply eslint-bridge/tests/rules/no-new-symbol.test.ts:84]: My bad, this overlapping should not have happened. The secondary location should refer to the `Symbol` identifier only, not the whole expression. I will fix that.

**Comment by yassin-kammoun-sonarsource** [comment]
> Isn't it valid also for typescript?

The TypeScript compiler considers `Symbol` in `new Symbol()` as a constructor. If it does not find such a constructor, the compiler raises an error. So I guess it's fine not to enable the rule for TypeScript.

**Review by andrea-guarino-sonarsource — APPROVED** [review]
LGTM, please merge when master is green and unfrozen

**Pull request merged**